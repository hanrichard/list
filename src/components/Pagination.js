import Grid from '@material-ui/core/Grid';
import Typography from '@material-ui/core/Typography';
import ArrowLeftIcon from '@material-ui/icons/ArrowLeft';
import ArrowRightIcon from '@material-ui/icons/ArrowRight';
import PropTypes from 'prop-types';
import React from 'react';

export default function Pagination({
  onClickPagination,
  currentPage,
  totalPaginatedPages,
}) {
  const handlePreviousClick = () => {
    if (currentPage > 1) {
      onClickPagination(currentPage - 1);
    }
  };

  const handleNextClick = () => {
    if (currentPage < totalPaginatedPages) {
      onClickPagination(currentPage + 1);
    }
  };
  return (
    <div data-testid="pagination">
      <Grid container justify="center">
        <Grid item xs>
          <ArrowLeftIcon
            data-testid="arrow-left-pagination"
            onClick={
              () => handlePreviousClick()
            } />
        </Grid>
        <Grid item xs={8}>
          <Typography variant="h4">
            Page
            {' '}
            {currentPage}
            /
            {totalPaginatedPages}
          </Typography>
        </Grid>
        <Grid item xs>
          <ArrowRightIcon
            data-testid="arrow-right-pagination"
            onClick={
              () => handleNextClick()
            } />
        </Grid>
      </Grid>
    </div>
  );
}

Pagination.defaultProps = {
  onClickPagination: () => {},
  currentPage: 0,
  totalPaginatedPages: 0,
};

Pagination.propTypes = {
  onClickPagination: PropTypes.func,
  currentPage: PropTypes.number,
  totalPaginatedPages: PropTypes.number,
};
