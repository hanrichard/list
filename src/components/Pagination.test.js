import { fireEvent, render } from '@testing-library/react';
import React from 'react';
import Pagination from './Pagination';

describe('<Pagination />', () => {
  it('should render accordingly', () => {
    const { asFragment } = render(
      <Pagination
        onClickPagination={jest.fn()}
        currentPage={1}
        totalPaginatedPagesNum={10} />,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('should call onClickPagination when clicking on the next pagination arrow', () => {
    const mockFn = jest.fn();
    const { getByTestId } = render(
      <Pagination
        onClickPagination={mockFn}
        currentPage={1}
        totalPaginatedPagesNum={10} />,
    );

    const rightArrowPagination = getByTestId('arrow-right-pagination');

    // Click next page
    fireEvent.click(rightArrowPagination);
    expect(mockFn).toBeCalledWith(2);
  });

  it('should call onClickPagination when clicking on the previous pagination arrow', () => {
    const mockFn = jest.fn();
    const { getByTestId } = render(
      <Pagination
        onClickPagination={mockFn}
        currentPage={2}
        totalPaginatedPagesNum={10} />,
    );

    const leftArrowPagination = getByTestId('arrow-left-pagination');

    // Click previous page
    fireEvent.click(leftArrowPagination);
    expect(mockFn).toBeCalledWith(1);
  });
});
