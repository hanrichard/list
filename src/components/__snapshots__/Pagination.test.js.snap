// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Pagination /> should render accordingly 1`] = `
<DocumentFragment>
  <div
    data-testid="pagination"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-justify-xs-center"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          data-testid="arrow-left-pagination"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M14 7l-5 5 5 5V7z"
          />
        </svg>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-8"
      >
        <h4
          class="MuiTypography-root MuiTypography-h4"
        >
          Page 1/10
        </h4>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          data-testid="arrow-right-pagination"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M10 17l5-5-5-5v10z"
          />
        </svg>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
